components/org.wso2.identity.outbound.adapter.websubhub/src/test/java/org/wso2/identity/outbound/adapter/websubhub/config/WebSubAdapterConfigurationTest.java

```diff
@@ -0,0 +1,69 @@
+package org.wso2.identity.outbound.adapter.websubhub.config;
+
+import org.mockito.Mockito;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.identity.outbound.adapter.common.OutboundAdapterConfigurationProvider;
+import org.wso2.identity.outbound.adapter.websubhub.exception.WebSubAdapterException;
+
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import static java.util.Objects.isNull;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertSame;
+import static org.testng.Assert.assertThrows;
+import static org.wso2.identity.outbound.adapter.websubhub.config.WebSubAdapterConfiguration.ADAPTER_ENABLED_CONFIG;
+import static org.wso2.identity.outbound.adapter.websubhub.config.WebSubAdapterConfiguration.ADAPTER_HUB_URL_CONFIG;
+import static org.wso2.identity.outbound.adapter.websubhub.config.WebSubAdapterConfiguration.ADPATER_HTTP_CLIENT_MAX_CONNECTIONS;
+import static org.wso2.identity.outbound.adapter.websubhub.config.WebSubAdapterConfiguration.ADPATER_HTTP_CLIENT_MAX_CONNECTIONS_PER_ROUTE;
+
+public class WebSubAdapterConfigurationTest {
+
+    @BeforeMethod
+    public void setup() {
+
+        String carbonConfPath = Paths.get(System.getProperty("user.dir"), "target", "test-classes").toString();
+        System.setProperty("carbon.config.dir.path", carbonConfPath);
+    }
+
+    @DataProvider(name = "websubAdapterPropertyProvider")
+    public String[][] provideSecurityEventData() {
+
+        return new String[][]{
+                // enabled, baseUrl, max_conn, max_conn_per_route
+                {null, null, null, null},
+                {"true", "http://localhost:9090/hub", "50", null},
+                {"true", null, "50", null},
+                {"false", "http://localhost:9090/hub", null, "25"},
+                {"false", null, "20", "25"}
+        };
+    }
+
+    @Test(dataProvider = "websubAdapterPropertyProvider")
+    public void testWebSubAdapterConfiguration(String... param) throws WebSubAdapterException {
```

Comment:
Can we use named parameters to improve readability.

Reply:
updated.